packages/zoe/src/zoeService/originalZoeSeat.js

```diff
@@ -0,0 +1,346 @@
+/* eslint @typescript-eslint/no-floating-promises: "warn" */
+import { SubscriberShape } from '@agoric/notifier';
+import { E } from '@endo/eventual-send';
+import { M, prepareExoClassKit } from '@agoric/vat-data';
+import { deeplyFulfilled } from '@endo/marshal';
+import { makePromiseKit } from '@endo/promise-kit';
+
+import { satisfiesWant } from '../contractFacet/offerSafety.js';
+import '../types.js';
+import '../internal-types.js';
+import {
+  AmountKeywordRecordShape,
+  KeywordShape,
+  ExitObjectShape,
+  PaymentPKeywordRecordShape,
+} from '../typeGuards.js';
+
+const { Fail } = assert;
+
+const OriginalZoeSeatIKit = harden({
+  zoeSeatAdmin: M.interface('ZoeSeatAdmin', {
+    replaceAllocation: M.call(AmountKeywordRecordShape).returns(),
+    exit: M.call(M.any()).returns(),
+    fail: M.call(M.any()).returns(),
+    resolveExitAndResult: M.call({
+      offerResultPromise: M.promise(),
+      exitObj: ExitObjectShape,
+    }).returns(),
+    getExitSubscriber: M.call().returns(SubscriberShape),
+    // The return promise is empty, but doExit relies on settlement as a signal
+    // that the payouts have settled. The exit publisher is notified after that.
+    finalPayouts: M.call(M.eref(PaymentPKeywordRecordShape)).returns(
+      M.promise(),
+    ),
+  }),
+  userSeat: M.interface('UserSeat', {
+    getProposal: M.call().returns(M.promise()),
+    getPayouts: M.call().returns(M.promise()),
+    getPayout: M.call(KeywordShape).returns(M.promise()),
+    getOfferResult: M.call().returns(M.promise()),
+    hasExited: M.call().returns(M.promise()),
+    tryExit: M.call().returns(M.promise()),
+    numWantsSatisfied: M.call().returns(M.promise()),
+    getFinalAllocation: M.call().returns(M.promise()),
+    getExitSubscriber: M.call().returns(M.any()),
+  }),
+});
```

Comment:
These seem identical with those in zoeSeat.js . Please de duplicate. If neither file should import the other, consider moving to a shared typeGuards.js . The commonality suggests that the interface is supposed to be the same.

Reply:
done